XenAdmin/Dialogs/NetworkingPropertiesPage.cs

```diff
@@ -166,15 +167,18 @@ public void RefreshButtons()
                 haEnabledRubric.Enabled =
                     true;
             }
-            if (network != null)
+
+            var pif = (PIF) Tag;

```

Comment:
Generally it's best to avoid implicit casting and do the following: `var pif = Tag as PIF;` The difference is that this one will not crash if the Tag were not to be a PIF.